miner/block_builder.go

```diff
@@ -0,0 +1,192 @@
+package miner
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/davecgh/go-xdr/xdr2"
+	"github.com/spacemeshos/go-spacemesh/common"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/state"
+	meshSync "github.com/spacemeshos/go-spacemesh/sync"
+	"math/big"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+const MaxTransactionsPerBlock = 200 //todo: move to config
+
+type BlockBuilder struct{
+	beginRoundEvent chan mesh.LayerID
+	stopChan		chan struct{}
+	newTrans		chan *state.Transaction
+	hareResult		chan HareResult
+	transactionQueue []SerializableTransaction
+	hareRes			HareResult
+	mu sync.Mutex
+	network p2p.Service
+	weakCoinToss	WeakCoinProvider
+	orphans			OrphanBlockProvider
+	started			bool
+}
+
+type Block struct {
+	Id         	mesh.BlockID
+	LayerIndex 	mesh.LayerID
+	Data       	[]byte
+	Coin       	bool
+	Timestamp  	time.Time
+	Txs			[]SerializableTransaction
+	HareResult	[]mesh.BlockID
```

Comment:
`Votes`? also shouldn't this be two slices for positive and negative result ? or a pair with voting direction and +/-.
also I don't see anything about view edges here.

Reply:
believe this one is solved as per @almogdepaz  comment below